xcp/pci.py

```diff
@@ -260,7 +260,7 @@ def __init__(self):
                                stdout = subprocess.PIPE)
         for l in cmd.stdout:
             line = l.rstrip()
-            el = [x for x in line.replace('"', '').split() if not x.startswith('-')]
```

Comment:
Thanks, this is a correct fix, because Popen() changes it's return value to byes.

Because the testsuite is not mocking this change in the return value of Popen() yet, this would break the test suite. It has to be changed like this:

tests/test_pci.py:
```diff
+        def fake_Popen():
+            """Mimic Popen().output: Python3 returns bytes, Python2 returns str"""
+            return open("tests/data/lspci-mn", "rb" if sys.version_info[0] == 3 else "r")
+
         with patch("xcp.pci.subprocess.Popen") as popen_mock, \
-             open("tests/data/lspci-mn") as fake_data:
+            fake_Popen() as fake_data:
             popen_mock.return_value.stdout.__iter__ = Mock(return_value=iter(fake_data))
             devs = PCIDevices()
```
I hope I can submit this tomorrow if you haven't by then.

Likely likewise for the Popen() for biosdevname, if it has a test.
If not, a test for it should be added.

Reply:
Hi @bernhardkaindl 
Thank you for helping me to review this PR, i will test with new branch.
Regards,
Tian

Reply:
Hi @bernhardkaindl 
Thank you for helping me to review this PR, i will test with new branch "testsuite-driven-py3+Popen".
Regards,
Tian

Reply:
The fixes issues which the changes in this PR adressed are now merged to master. 

Hi @alexhimmel, you could schedule re-testing the ACK XAPI plugin with the new master branch in an upcoming sprint!

PS: The Python3 migration is not done yet, there are still many issues (or at least untested cases) which need testing and likely fixing.

For example the github action summay page is currently still showing pytype errors in xcp/accessor.py and xcp/repository.py.

If the ACK plugin uses such modules or you run into any other error, please open an issue here to report the problem, thanks!